develop/tutorials/articles/310-troubleshooting/03-resolving-cnfe-and-cdnfe-in-osgi-bundles.markdown

```diff
@@ -0,0 +1,72 @@
+# Resolving ClassNotFoundException and NoClassDefFoundError in OSGi Bundles [](id=resolving-classnotfoundexception-and-noclassdeffounderror-in-osgi-bundles)
+
+Understanding a `ClassNotFoundException` or `NoClassDefFoundError` in non-OSGi
+environments is straightforward. 
+
+-   `ClassNotFoundException`: thrown when looking up a class that isn't in the
+    classpath or using an invalid name to look up a class.
+-   `NoClassDefFoundError`: occurs when a compiled class references
+    another class that isn't in the run time classpath.
+
+In OSGi environments, there are additional cases in which a
+`ClassNotFoundException` or `NoClassDefFoundError` can occur. Here are two
+common cases:
+
+1.  Bundle doesn't import the class' package.
```

Comment:
After "Screens's," I thought you would have had this correct by now, @ngaskill. :-)